### PR TITLE
 Error display in logs when the generator is unable to send email.

### LINKED ISSUE #1450: Logs display 'Mail sent succesfully' even if error message is received on sending email.
**Actual Behaviour**

Logs display 'Mail Sent succesfully' even if error like proxy issue or network error is received.

**Expected Behaviour**

If 200 OK is not received in email response logs should display 'Mail not sent'

**Steps to reproduce it**
On your local machine make the port settings incompatible to send email,try generating app logs will display success message.

**Log for the issue**
Error sending Email { SendGridError: Response error
    at /home/ayush/WebstormProjects/openEventWebappMaster/open-event-webapp/node_modules/sendgrid/lib/sendgrid.js:104:23
    at ClientRequest.<anonymous> (/home/ayush/WebstormProjects/openEventWebappMaster/open-event-webapp/node_modules/sendgrid-rest/lib/client.js:120:7)
    at emitOne (events.js:115:13)
    at ClientRequest.emit (events.js:210:7)
    at TLSSocket.socketErrorListener (_http_client.js:401:9)
    at emitOne (events.js:115:13)
    at TLSSocket.emit (events.js:210:7)
 at emitOne (events.js:115:13)
    at emitErrorNT (internal/streams/destroy.js:64:8)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickDomainCallback (internal/process/next_tick.js:218:9)
  message: 'Response error',
  response: 
   { statusCode: 500,
     body: 
      { message: 'connect ECONNREFUSED 167.89.125.80:443',
        name: 'Error',
        stack: 'Error: connect ECONNREFUSED 167.89.125.80:443\n    at Object._errnoException (util.js:1041:11)\n    at _exceptionWithHostPort (util.js:1064:20)\n    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1153:14)' },
     headers: {} } }

**Screenshots of the issue**

screenshot from 2017-08-19 23-47-31
screenshot from 2017-08-19 23-47-41

**Would you like to work on the issue?**
Yes I would like to work on this issue.

### PR DESCRIPTION
**Pull request by agbilotia1998**
Fixes #1450 

Changes: Added if statement in generator.js for correctly displaying the logs.

Screenshots for the change: 
screenshot from 2017-08-20 01-24-56

### CODE REVIEW

**Comment by Princu7** [comment]
Thanks for the PR :grinning: Good work!
While we are at it, there are some more things to consider here. This is the workflow
* We generate the event folder and then upload to the cloud storage and get the URL of the uploaded folder.
* We then mail the user about the generated event which also contains a link to the uploaded event folder.

The question is what should we do when the upload to the cloud upload fails? Directly sending the mail to the user does not make sense since the download location of the uploaded event folder will be null.

So, we have two options maybe:-

* Don't send a mail to the user if the cloud upload fails.
* Send a different email telling that the cloud upload has failed and present the user the download link of the event folder currently present on the Heroku app itself along with the warning that it will expire within few hours.

Thoughts @aayusharora @geekyd @sumedh123 @harshitagupta30 I also think that it might be out of the scope of this issue so we can open a new issue for it I guess.

**Comment by sumedh123** [comment]
@Princu7 The second option looks good to me.

**Comment by Princu7** [comment]
@agbilotia1998 Don't worry much about the failing tests. Code coverage measure how much portion of the code is covered by the existing tests (In simpler terms, how much portion of the code is executed during the tests). 
See these couple of lines from the commit
```
if(url)
     logger.addLog('Success', 'Mail sent succesfully', socket);
```
The second line is never executed since the mail is never send to the user (Travis doesn't have the appropriate keys). So, this line is accounted as a miss which leads to the failing tests. But it is very small and negligible change so it is almost harmless.

**Comment by agbilotia1998** [comment]
Okay @Princu7.Thanks.Any changes to be done with this PR?

**Comment by Princu7** [comment]
Yes. I will open a new issue regarding the problem I have mentioned. For this issue, this commit should be enough I guess. Just make some changes

* Don't write 'Fixes #issueNo' in the commit message. Just write the former in the PR description. Not important at all to write in the commit. Something like 'Make the error display in logs when the generator is unable to send email' should be fine.
* Update the PR with the latest changes in the development branch.
* Squash the commits into one

Just do these changes and it should be fine. Thanks :+1:

**Comment by agbilotia1998** [comment]
@Princu7 done,please review :)
